freediscovery/interpretation.py

```diff
@@ -0,0 +1,170 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import re
+import matplotlib as mpl
+import matplotlib.cm as cm
+import numpy as np
+
+
+def explain_logistic_regression(weights, textlines):
+    """
+    Generate decorated HTML by adding <span> tags with background color to keywords depending on word's weight.
+
+    Parameters
+    ----------
+    weights: [dict] {vocabulary : weights} with words [str] as keys, weights (scores) [float] as values
+    textlines [list of str] list of text lines representing raw document text
+
+    Returns
+    -------
+    [str] HTML string representing document's text decorated with <span> tags
+    """
+    document_html = ''
+    for line in textlines:
+        line = line.replace('<', '',).replace('>', '')  # in order to have a valid HTML output after adding <span> tags

```

Comment:
Right, I see the problem..  Let's rather  html encode all the text,
```py
try:
    from html import escape  # python 3.x
except ImportError:
    from cgi import escape  # python 2.x
text = escape(text)
```
as "&" and some other characters might also be an issue here...